src/WebJobs.Extensions.DocumentDB/Config/DocumentDBConfiguration.cs

```diff
@@ -80,7 +80,7 @@ public void Initialize(ExtensionConfigContext context)
             IExtensionRegistry extensions = context.Config.GetService<IExtensionRegistry>();
             extensions.RegisterBindingRules<DocumentDBAttribute>(ValidateConnection, nameResolver, outputProvider, clientProvider, jArrayProvider, enumerableProvider, inputProvider);
 
-            context.Config.RegisterBindingExtensions(new CosmosDBTriggerAttributeBindingProvider(ResolveConnectionString(null), ResolveConnectionString(null), LeaseOptions));
+            context.Config.RegisterBindingExtensions(new CosmosDBTriggerAttributeBindingProvider(nameResolver, ResolveConnectionString(null), ResolveConnectionString(null), LeaseOptions));
```

Comment:
Since ultimately what this is trying to do is make sure both the monitor and lease connection strings fall back to the default, I think we should bake that logic into the BindingProvider itself. It'll allow you to test the fallbacks more easily (see other comments). Can we remove the 2nd and 3rd parameters here? Then in TryCreateAsync you can do:

```
_monitorConnectionString = _nameResolver.Resolve(DocumentDBConfiguration.AzureWebJobsDocumentDBConnectionStringName);
_leasesConnectionString = _nameResolver.Resolve(DocumentDBConfiguration.AzureWebJobsDocumentDBConnectionStringName);
```